pages/users/[id]/index.js

```diff
@@ -0,0 +1,87 @@
+import { useRouter } from "next/router";
+import styled from "styled-components";
+import useSWR from "swr";
+import { StyledList } from "@/styles";
+import ActivityCard from "@/components/ActivityCard";
+import { useSession } from "next-auth/react";
+
+export default function UserPage({ activityData, onToggleFavorite, userData }) {
+  const router = useRouter();
+  const { id } = router.query;
+  const { data: userProilData } = useSWR(`/api/users/${id}`);
+  const { data: session } = useSession();
+
+  if (!userProilData) return <p>Loading...</p>;
+
+  return (
+    <>
+      <UserImage
+        src={userProilData.picture || "/images/user_picture.png"}
+        alt={`${userProilData.name}'s profile picture`}
+      />
+      <UserName>{userProilData.name}</UserName>
+      <UserLocation>
+        {userProilData.city && userProilData.country
+          ? `${userProilData.city}, ${userProilData.country}`
+          : ""}
+      </UserLocation>
+      <UserAbout>
+        <SectionTitle>About Me</SectionTitle>
+        <AboutText>{userProilData.aboutMe || "No details provided."}</AboutText>
+      </UserAbout>
+      <StyledList>
+        {activityData
+          .filter((activity) => activity.owner === id)
+          .map((userActivity) => (
+            <li key={userActivity._id}>
+              <ActivityCard
+                key={userActivity._id}
+                activity={userActivity}
+                onToggleFavorite={onToggleFavorite}
+                isFavorite={
+                  session
+                    ? (userData?.favorites ?? []).includes(userActivity._id)
+                    : false
```

Comment:
```suggestion
                    && (userData?.favorites ?? []).includes(userActivity._id)
```
If session is undefined, this will return "false" as well.